test/integration/node-specific/auto_connect.test.ts

```diff
@@ -862,12 +863,17 @@ describe('When executing an operation for the first time', () => {
           sinon.restore();
         });
 
-        it('client.connect() takes as long as selectServer is delayed for and does not throw a timeout error', async function () {
+        it('client.connect() takes as long as selectServer is delayed for and throws a timeout error', async function () {
           const start = performance.now();
           expect(client.topology).to.not.exist; // make sure not connected.
-          const res = await client.db().collection('test').insertOne({ a: 1 }, { timeoutMS: 500 }); // auto-connect
+          const error = await client
+            .db()
+            .collection('test')
+            .insertOne({ a: 1 }, { timeoutMS: 500 })
+            .catch(error => error);
```

Comment:
This is because the operation's server selection is the only one that's running, and 500ms has expired by the time we reach it. 

Not sure what the best path forward for this is 🤔